crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -2721,6 +2738,23 @@ impl<F: Clone> PostUpdateTracker<F, PaymentStatusData<F>, types::PaymentsSyncDat
 
         let response_router_data = response;
 
+        if let (true, true, Some(payment_method_id)) = (
```

Comment:
move to domain and call the domain function in core flow instead adding this in trackers